### PR TITLE
SLIM-706 Enhance error reporting in bundled responses

### PR DESCRIPTION
**Pull request by bvdzwet**

### CODE REVIEW

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java [hunk 1]
```diff
@@ -64,12 +68,59 @@ protected void handleError(final Exception e, final DeviceMessageMetadata device
...
         }
 
         this.bundleService.handleBundleResponse(deviceMessageMetadata, responseMessage.getResult(), osgpException,
                 bundleMessagesResponseDto);
     }
+
+    private FaultResponseDto faultResponseForException(final Exception exception,
+            final List<FaultResponseParameterDto> parameters, final String defaultMessage) {
+
+        if (exception instanceof FunctionalException || exception instanceof TechnicalException) {
```
Comment (robinbakkeruscgi) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java]: Is there a special reason why not any OsgpException is used?

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java [hunk 2]
```diff
@@ -64,12 +68,59 @@ protected void handleError(final Exception e, final DeviceMessageMetadata device
...
-                        new ActionResponseDto(OsgpResultTypeDto.NOT_OK, e, "Unable to handle request"));
+                        this.faultResponseForException(e, parameterList, "Unable to handle request"));
             }
         }
 
         this.bundleService.handleBundleResponse(deviceMessageMetadata, responseMessage.getResult(), osgpException,
                 bundleMessagesResponseDto);
     }
+
+    private FaultResponseDto faultResponseForException(final Exception exception,
```
Comment (robinbakkeruscgi) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java]: The code below including faultResponseForFunctionalOrTechnicalException, imo deserves a jUnit test

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/FaultResponse.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.alliander.osgp.domain.core.valueobjects.smartmetering;
+
+import java.util.Objects;
+
+public class FaultResponse extends ActionResponse {
```
Comment (robinbakkeruscgi) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/FaultResponse.java]: There are no mappingTest(s) needed for these new value objects !?

Comment (bvdzwet) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/FaultResponse.java]: Wouldn't this be just testing that the Orika mapping library does what it should do? I didn't think I added converters for these objects doing anything to deserve special attention.
The fact this does not go wrong will be apparent as soon as we know more details about how and when to use the fault responses from integration tests (this PR includes just the set up that will make further steps possible going forward).

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java (hunk 1 above)
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java]: The two OsgpExceptions checked for explicitly are the only ones that were converted to details in SOAP faults with the single platform requests (see e.g. `FunctionalExceptionConverter` or `TechnicalExceptionConverter` in osgp-adapter-ws-smartmetering). For the fault responses in bundles these details are mimicked to preserve relevant information appearing in a "regular" response.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java (hunk 2 above)
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/BundleResponseMessageProcessor.java]: Some unit tests for creation of `FaultResponseDto` objects have been added.

**Pull request merged**